LandSandBoat/server
# Fix issue with Indi-type spell targeting

**_I affirm:_**

## What does this pull request do?

Sets Indi-type spells to have TARGET_SELF flag and gets rid of some now unnecessary checks (CBattleEntity::ValidTarget handles self targeting). This could maybe be handled in the same check as pianissimo, but that also has some pet checks that I'm not sure about.

## Steps to test these changes

```
!addspell 981
!addspell 768
!changejob war 99
```
Cast Sylvie trust (no log errors), engage a target. She should Fury herself then entrust Frailty to you.
```
!changejob bst 99
!additem 17860
```
Equip the jug, call beast, use entrust, target pet, try casting indi spell on pet.
